Source/VCGeneration/SplitAndVerifyWorker.cs

```diff
@@ -100,16 +102,17 @@ async Task DoWork(Split split, CancellationToken cancellationToken)
 
       try {
         cancellationToken.ThrowIfCancellationRequested();
-        StartCheck(split, checker, cancellationToken);
+        var splitCallback = new SplitVerifierCallback(callback, split);
```

Comment:
Could you move the logic for recording which CounterExamples a Split has into the `Split` class? `Split` creates a `VCGen.ErrorReporter` where this callback is also passed.

Reply:
I wish I could move this logic, however, it would require me to also move the logic for handling the split's result into the split itself, since `SplitAndVerifyWorker.HandleProverFailure` and `SplitAndVerifyWorker.ProcessResult` both rely on the `SplitVerifierCallback`. Let me know if you would like me to attempt such a refactoring as part of this PR, or if it is fine for now.

Reply:
Looking at your solution, I was able to find another one that did not even need to change the line you pointed at.
The class SplitVerifierCallback is now removed !